docs/source/how_to/how_to_algorithm_selection.ipynb

```diff
@@ -35,34 +63,92 @@
    "source": [
     "import numpy as np\n",
     "\n",
-    "import optimagic as om"
+    "import optimagic as om\n",
+    "\n",
+    "\n",
+    "def trid_scalar(x):\n",
+    "    \"\"\"Implement Trid function: https://www.sfu.ca/~ssurjano/trid.html.\"\"\"\n",
+    "    return ((x - 1) ** 2).sum() - (x[1:] * x[:-1]).sum()\n",
+    "\n",
+    "\n",
+    "def trid_gradient(x):\n",
+    "    \"\"\"Calculate gradient of trid function.\"\"\"\n",
+    "    l1 = np.insert(x, 0, 0)\n",
+    "    l1 = np.delete(l1, [-1])\n",
+    "    l2 = np.append(x, 0)\n",
+    "    l2 = np.delete(l2, [0])\n",
+    "    return 2 * (x - 1) - l1 - l2"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "@om.mark.least_squares\n",
-    "def sphere(params):\n",
-    "    return params\n",
+    "## Step 1: Theory\n",
+    "\n",
+    "The below decision tree offers a practical guide on how to narrow down the set of algorithms to experiment with, based on the theoretical properties of your problem:\n",
+    "\n",
+    "```{mermaid}\n",
+    "graph LR\n",
+    "    classDef highlight fill:#FF4500;\n",
+    "    A[\"Do you have<br/>nonlinear constraints?\"] -- yes --> B[\"differentiable?\"]\n",
```

Comment:
Currently, it also seems important how many bounds and constraints there are to avoid the issue #541 described. 

(But not sure whether and how it makes sense to mention it)

Reply:
We only consider nonlinear constraints (which are handled by some optimizers directly). Other types of constraints are not relevant for the choice because they are handled via reparametrization. I would leave it out of this guide.